tls/s2n_cipher_suites.c

```diff
@@ -1168,11 +1171,164 @@ static int s2n_wire_ciphers_contain(const uint8_t *match, const uint8_t *wire, u
     return 0;
 }
 
+S2N_RESULT s2n_cipher_suite_uses_chacha20_alg(struct s2n_cipher_suite* cipher_suite, bool* uses_chacha20_alg) {
+    RESULT_ENSURE_REF(cipher_suite);
+    RESULT_ENSURE_REF(uses_chacha20_alg);
+
+    *uses_chacha20_alg = false;
+
+    for (size_t i = 0; i < cipher_suite->num_record_algs; i++) {
+        const struct s2n_record_algorithm* alg = cipher_suite->all_record_algs[i];
+        if (alg == &s2n_record_alg_chacha20_poly1305 || alg == &s2n_tls13_record_alg_chacha20_poly1305) {
+            *uses_chacha20_alg = true;
+            return S2N_RESULT_OK;
+        }
+    }
+
```

Comment:
This is unnecessary. After the ciphers have been setup during init, you can just do cipher_suite->record_alg. Beyond that, you can then do cipher_suite->record_alg->cipher to check for the single implementation of chacha20.

Reply:
Thanks